.github/workflows/osv-scanner-pr.yml

```diff
@@ -0,0 +1,106 @@
+# This file is adapted from https://github.com/google/osv-scanner
+
+name: OSV-Scanner PR scanning
+
+permissions: {}
+
+on:
+  workflow_call:
+    inputs:
+      scan-args:
+        description: "Custom osv-scanner arguments (See https://google.github.io/osv-scanner/usage/ for options, you cannot set --format or --output)"
+        type: string
+        default: |-
+          -r
+          --skip-git
+          ./
+      results-file-name:
+        description: "File name of the result SARIF file"
+        type: string
+        default: results.sarif
+      upload-sarif:
+        description: "Whether to upload to Security > Code Scanning"
+        type: boolean
+        required: false
+        default: true
+      fail-on-vuln:
+        description: "Whether to fail the action on vulnerability found"
+        type: boolean
+        default: true
+
+jobs:
+  scan-pr:
+    permissions:
+      contents: read # to fetch code (actions/checkout)
+      security-events: write # for uploading SARIF files
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      with:
+        fetch-depth: 0
+        # Do persist credentials, as we need it for the git checkout later
+    - name: "Checkout target branch"
+      run: git checkout $GITHUB_BASE_REF
+    - name: "Calculate go version"
```

Comment:
@tuminoid check it, I am not calculating the go version from makefile and setting up the go in the next action before running OSV scanner run to make sure we have the correct go patch version in the env, is there a way to verify if this is being used?

Reply:
Sure, I'll try if the `gh act` can be used to play with this workflow.

Reply:
It seems to work fine.